### DISCUSSION TITLE
Eloquent query results are returning all translations

### DISCUSSION THREAD

**Discussion by binaryfire** [spatie/laravel-translatable#433] [Q&A]
Hi all

Not sure if I'm doing something wrong, but when I get multiple results using Eloquent's `get()` or `all()` I'm getting all the translations. Is that supposed to be happening? I expected that the package would just return the current locale's translation for the attribute

**Comment by youseffatouraee** [comment]
Your expectation is correct.

Example:
```php
$apples = Apple::all();
foreach ($apples as $apple) {
        echo $apple->name;
}
```
This should echo all the translated values.

**Reply by binaryfire** [reply]
Hi @ModulesSoft. So for you , that just echos the translated values for the app's current locale? Not the translations for all locales?

**Reply by youseffatouraee** [reply]
Yes.
In order to get all the translations, you can use the following method:
`
$newsItem->getTranslations();
`
See:
https://spatie.be/docs/laravel-translatable/v6/basic-usage/getting-and-settings-translations#content-getting-all-translations

Your problem might be with your models or database records for translatable columns. 
Check your `$translatable` property in your models and if your database records are set and structured correctly.

**Reply by youseffatouraee** [reply]
Was it helpful?

**Comment by Studio384** [comment]
I am facing a similar issue on Laravel 11 with the latest laravel-translate, using MariaDB 11.3. With a model setup like this:

```php
class FaqQuestion extends Model
{
    use HasTranslations;

    protected $table = 'faq_questions';

    protected $fillable = [
        'question',
        'answer',
        'category_id',
    ];

    public $translatable = ['question', 'answer'];
}
```

With the following migration:

```php
Schema::create('faq_questions', function (Blueprint $table) {
    $table->id();
    $table->json('question');
    $table->json('answer');
    $table->foreignId('category_id')->constrained()->cascadeOnDelete();
    $table->timestamps();
});
```

Calling `FaqQuestion::find(1)` will result in this response:

```php
[
    "id" => 1
    "question" => "{"nl":"Wat is Nederlands?","en":"What is English?"}"
    "answer" => "{"nl":"<p>Nederlands antwoord.</p>","en":"<p>English answer.</p>"}"
    "category_id" => 1
    "created_at" => "2025-02-03 08:22:01"
    "updated_at" => "2025-02-03 08:22:01"
]
```

All available languages are provided rather than just the current locale (or fallback). Is there a configuration I could've missed here?
